app/assets/javascripts/Components/Result/file_viewer.jsx

```diff
@@ -111,17 +128,25 @@ export class FileViewer extends React.Component {
       this.remove();
     });
 
-    this.setState({loading: true, url: null}, () => {
+    this.setState({loading: true, size: 0, url: null}, () => {
       if (!this.props.selectedFileURL) {
         fetch(
           Routes.get_file_course_submission_path(this.props.course_id, this.props.submission_id, {
             submission_file_id: submission_file_id,
             force_text: force_text,
+            max_size: this.max_size,
```

Comment:
There is no attribute called `max_size`, though there's one called `max_sizes`, which is an object, not a number.

Your current implementation only uses the size parameter for the plaintext case, so you could pass in just a number `max_text_size` here and use it in the controller method in the appropriate branch.